### PR TITLE
Fix the garbled file name

### PR DESCRIPTION
**Pull request by CNLouisLiu**
Fixed the problem of garbled file names with non-ASCII characters
bac3a5f0c30fc6f12d9da3bb1361d4d9

### CODE REVIEW

File: Dumper/Utils.h
Comment (Fischsalat) [inline Dumper/Utils.h:1]: There's already a whole library for Unicode chars, no need to use deprecated c++17 stuff.

File: Dumper/CppGenerator.cpp
Comment (Fischsalat) [inline Dumper/CppGenerator.cpp:1]: This diff is impossible to look at, either make two different commits, one with formatting, one with the actual changes, or leave out the formatting part entirely.

**Comment by Fischsalat** [comment]
Which file names are "garbled"? What did you do to fix that? What game does this happen on?

**Comment by CNLouisLiu** [comment]
> Which file names are "garbled"? What did you do to fix that? What game does this happen on?

The class name uses non-ascii characters, and the file is written out in the default way, that is, utf8string. Unicode file names should be used when writing out the file.
